### DISCUSSION TITLE
Vanish Status without essentials.vanish permission resets on world change

### DISCUSSION THREAD

**Discussion by molor** [EssentialsX/Essentials#4817] [Ideas and Feature Suggestions]
_don't want to fill an issue with I think much useless info, and also don't think anyone will answer at all, looking at the number of unconfirmed and unanswered issues... so I just created a discussion instead, in case it's intended behavior_

If server operator will change the vanish status of another player to `on`, this status will be reverted back to `off` after world change.
I think that this is a bug, at least there's no any disabling notice in the chat so player may just don't know that others can see it now.

Can be easily reproduced by using nether portal or just /spawn from another world.

Currently adding `essentials.vanish` to player's permission list solves this issue. But another permission like `essentials.vanish.keep` would be better.. If that's not a bug, add an notification message in chat please.

Thanks for your time and work.

```
[15:01:38] Server version: 1.18.1-R0.1-SNAPSHOT git-Paper-214 (MC: 1.18.1) (fine)
[15:01:38] Brand version: Paper (fine)
[15:01:38] EssentialsX version: 2.19.3-dev+27-bc2da98 (fine)
[15:01:38] LuckPerms version: 5.4.0 (fine)
[15:01:38] Vault version: 1.7.3-b131 (fine)
[15:01:38] EssentialsXGeoIP version: 2.19.3-dev+27-bc2da98 (fine)
[15:01:38] EssentialsXSpawn version: 2.19.3-dev+27-bc2da98 (fine)
```

**Comment by mdcfe** [comment]
We could add some sort of persistent mode when `/vanish`ing other players, allowing vanish status to persist even when players change into worlds without permission. However, I'm not sure whether this is really within scope for EssentialsX's vanish feature.

Another alternative is to add a config option allowing server owners to disable the world change check, similar to `world-change-fly-reset`. I would be open to a PR that adds this.

**Reply by kennyrkun** [reply]
I think it's definitely within the scope of the vanish feature to keep a player (regardless of their own permissions) vanished up until the user is explicitly removed from the vanish state by way of the command being issued again.
